.github/workflows/codeql-build.yml

```diff
@@ -1,37 +1,23 @@
 name: "CodeQL Analysis"
 
+# Only trigger, when the build workflow succeeded
 on:
-  push:
-  pull_request:
-    branches:
-      - main
-
+  workflow_run:
+    workflows: 
+      - Build, Test, and Run
```

Comment:
You can use the syntax - ["WORKFLOW_NAME_WITH_SPACES"] 
See https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_run

Reply:
Tested: 

Build, Test, and Run [OMIT_DEPRECATED=true] https://github.com/ArielSAdamsNASA/cFS-JSF-Rules/actions/runs/1046324868

CodeQL was not triggered: https://github.com/ArielSAdamsNASA/cFS-JSF-Rules/actions/workflows/codeql-build.yml